### PR TITLE
	    QPPSF-5794: Broken ECQM measure links fix

### PR DESCRIPTION
**Pull request by mssb-hub**
Motivation for change

                     User facing interface is currently showing dead links. The fix will provide relevant information and better user experience.

                 What is being changed

                     index.js
	             measures/2017/measures-data.json
		     measures/2018/measures-data.json
		     test/measures/measures-specification-spec.js

                 Release checklist:
             Tasks that must be done prior to merging this PR, including testing.

            * [x] Unit tests added/passing
            * [x] Verified working locally

                  Associated JIRA tickets:
            * https://jira.cms.gov/browse/QPPSF-5794

Include any related JIRA issue keys in the PR title, eg `QPPA-1234 Add measures x, y, and z`.

#### Motivation for change

User facing interface is currently showing dead links. The fix will provide relevant information and better user experience.

#### What is being changed

                     index.js
	             measures/2017/measures-data.json
		     measures/2018/measures-data.json
		     test/measures/measures-specification-spec.js

#### Release checklist:
Tasks that must be done prior to merging this PR, including testing.

* [ ] [Package version updated](https://github.com/CMSgov/qpp-measures-data/blob/master/CONTRIBUTING.md#versioning-publishing-and-creating-new-releases)
* [ ] Documentation updated
* [ ] Unit tests added/passing
* [ ] Verified working locally

##### Associated JIRA tickets:
* https://jira.cms.gov/browse/QPPA-5794

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,6 +47,15 @@ exports.getMeasuresSchema = function(performanceYear = 2017) {
   return YAML.load(path.join(__dirname, 'measures', performanceYear.toString(), 'measures-schema.yaml'));
 };
 
+
+/**
+ * @return {Array<Measure>}
+ */
+exports.getMeasuresDataByPerfYear = function(performanceYear) {
```
Comment (branonbarrett) [inline index.js]: This function is exactly the same as `exports.getMeasuresData = function(performanceYear = 2017) {` on line 38 and as such is unnecessary.

File: test/measures/measures-specification-spec.js [hunk 2]
```diff
@@ -2,53 +2,94 @@ const chai = require('chai');
...
+
+  measures
+    .map(m => ({measureId: m.measure.measureId, measureSpecification: m.measure.measureSpecification,
+      firstPerformanceYear: m.measure.firstPerformanceYear, performanceYear: m.performanceYear,
+      eMeasureId: m.measure.eMeasureId, title: m.measure.title, nqfId: m.measure.nqfId
+    }))
+    .filter(s => !!s.measureSpecification)
+    .forEach(s => {
+      if(typeof s.measureSpecification === 'object'){
+        /*Object.values(s.measureSpecification).forEach(url => {
```
Comment (branonbarrett) [inline test/measures/measures-specification-spec.js]: Remove commented out code

File: test/measures/measures-specification-spec.js [hunk 3]
```diff
@@ -2,53 +2,94 @@ const chai = require('chai');
...
+  return Promise.map(specs, s => checkUrl(s), { concurrency: 20 })
+    .then(results => {
+      const failures = results.filter(r => !r.success);
+      if (failures.length > 0) {
+        console.log(failures);
+        fs.writeFileSync( 'test/measures/measure-link-failures.txt',  JSON.stringify(failures, null, 2));
+      }
+      assert.equal(0, failures.length, 'One or more measure specifications link is invalid');
+    })
+  //)
```
Comment (branonbarrett) [inline test/measures/measures-specification-spec.js]: Remove commented out code

File: test/measures/measures-specification-spec.js [hunk 4]
```diff
@@ -2,53 +2,94 @@ const chai = require('chai');
...
+});
+
+//}
```
Comment (branonbarrett) [inline test/measures/measures-specification-spec.js]: same

**Comment by joshboley** [comment]
Were the measure json files manually updated?  If so, these changes will be overwritten whenever someone makes updates in the future.  We need to update the source data.  All measure json files are generated by code.

**Comment by joshboley** [comment]
Closing as this was done in #327

**Pull request closed without merging**